verticapy/learn/delphi.py

```diff
@@ -0,0 +1,835 @@
+        specificity : Specificity = tn / (tn + fp)
+    For Regression:
+        max    : Max Error
+        mae    : Mean Absolute Error
+        median : Median Absolute Error
+        mse    : Mean Squared Error
+        msle   : Mean Squared Log Error
+        r2     : R squared coefficient
+        r2a    : R2 adjusted
+        rmse   : Root Mean Squared Error
+        var    : Explained Variance 
+cv: int, optional
+    Number of folds.
+pos_label: int/float/str, optional
+    The main class to be considered as positive (classification only).
+cutoff: float, optional
+    The model cutoff (classification only).
+nbins: int, optional
+    Number of bins used to compute the different parameters categories.
+lmax: int, optional
+    Maximum length of each parameter list.
+optimized_grid: int, optional
+    If set to 0, the randomness is based on the input parameters.
+    If set to 1, the randomness is limited to some parameters, the other
+    ones are picked based on a default grid.
+    If set to 2, there is no randomness and a default grid is returned.
+stepwise: bool, optional
+    If set to True, the stepwise algorithm will be used to determine the
+    final model list of parameters.
+stepwise_criterion: str, optional
+    Criterion used when doing the final estimator stepwise.
+        aic : Akaike’s Information Criterion
+        bic : Bayesian Information Criterion
+stepwise_direction: str, optional
+    How to start the stepwise search. Can be done 'backward' or 'forward'.
+stepwise_max_steps: int, optional
+    The maximum number of steps to be considered when doing the final estimator 
+    stepwise.
+x_order: str, optional
+    How to preprocess X before using the stepwise algorithm.
+        pearson  : X is ordered based on the pearson correlation coefficient.
+        spearman : X is ordered based on the spearman correlation coefficient.
+        random   : Shuffles the vector X before applying the stepwise algorithm.
+        none     : Does not change X order.
+preprocess_data: bool, optional
+    If set to True, the data will be preprocessed.
+preprocess_dict: dict, optional
+    Dictionary to pass to the AutoDataPrep class in order to 
+    preprocess the data before the clustering.
+print_info: bool
+    If set to True, prints the model information at each step.
+
+Attributes
+----------
+preprocess_: object
+    Model used to preprocess the data.
+best_model_: object
+    Most relevant model found during the search.
+model_grid_ : tablesample
+    Grid containing the different models information.
+    """
+    # ---#
+    def __init__(self,
+                 name: str,
+                 cursor=None,
+                 estimator: (list, str) = "fast",
+                 estimator_type: str = "auto",
+                 metric: str = "auto",
+                 cv: int = 3,
+                 pos_label: (int, float, str) = None,
+                 cutoff: float = -1,
+                 nbins: int = 1000,
```

Comment:
The default value of `nbins` seems to be very big. I wonder what has been your criteria for picking the default values.

Reply:
Done; I put it to 100, I changed it for testing purpose.